crates/storage/provider/src/block.rs

```diff
@@ -168,15 +166,25 @@ pub fn insert_canonical_block<'a, TX: DbTxMut<'a> + DbTx<'a>>(
         let rec_tx = transaction.clone().into_ecrecovered().unwrap();
         tx.put::<tables::TxSenders>(current_tx_id, rec_tx.signer())?;
         tx.put::<tables::Transactions>(current_tx_id, rec_tx.into())?;
+        transition_id += 1;
         tx.put::<tables::TxTransitionIndex>(current_tx_id, transition_id)?;
         current_tx_id += 1;
-        transition_id += 1;
     }
```

Comment:
this means we would start indexing transitions from index `1`, no?

Reply:
Same Q

Reply:
It should start from `0`! 
I was sure that change was solid as I just switched the order of increment from `last_transition_id+1`  to before `tx.put`.
But I think there is a small bug in both body stage and here, it is more semantics and does not do anything major;

If we have block without transaction ``last_transition_id+1` would increment the transition and leave a gap and block reward would do the same.

Here is the summary code: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ea0f8335487dc0ccae4519cb0abd2294

And in body stage we are doing it here: https://github.com/paradigmxyz/reth/blob/06be083335e3fa6106bb13ac00e0b02610fa936d/crates/stages/src/stages/bodies.rs#L96

and get_next_block_id is incrementing it here:
https://github.com/paradigmxyz/reth/blob/06be083335e3fa6106bb13ac00e0b02610fa936d/crates/stages/src/db.rs#L176

Reply:
Here is the commit that resolves it

Reply:
@rkrasiuk when you can, can you confirm this?